File: Gridly-Client/src/app/Models/Card.Model.ts
import {ComponentSettingsModel} from "./ComponentSettings.Model";
import {IconModel} from "./Icon.Model";

export class CardModel {
  id!: number;
  indexPosition!: number;
  name!: string;
  url!: string;
  iconData?: IconModel |
    {
      type: "",
      name: "",
      base64Data: "",
      materialIcon: ""
    };
  iconUrl?: string;
  materialIcon?: string;
  componentSettings? : ComponentSettingsModel |
    {
      width:250,
      height:250,
      imageHidden:false,
      titleHidden:false
    };
}

File: Models/CardModel.cs
using System.Text.Json.Serialization;

namespace Gridly.Models;

public class CardModel
{
    [JsonPropertyName("id")] public int Id { get; set; }
    [JsonPropertyName("indexPosition")] public int? IndexPosition { get; set; }
    [JsonPropertyName("name")] public string? Name { get; set; }
    [JsonPropertyName("url")] public string? Url { get; set; }
    [JsonPropertyName("iconData")] public IconModel? IconData { get; set; }
    [JsonPropertyName("iconUrl")] public string? IconUrl { get; set; }
    [JsonPropertyName("cardSettings")] public SettingsModel? Settings { get; set; }
}
